nemo/core/classes/modelPT.py

```diff
@@ -606,10 +606,44 @@ def setup_optimizer_param_groups(self):
             See https://pytorch.org/docs/stable/optim.html for more information.
             By default, ModelPT will use self.parameters().
             Override this method to add custom param groups.
-    """
-        param_groups = None
-        if hasattr(self, 'parameters'):
-            param_groups = [{'params': self.parameters()}]
+            In the config file, add 'optim_param_groups' to support different LRs 
+            for different components (unspecified params will use the default LR):
+            model:
+                optim_param_groups:
+                    encoder: 1e-4
+                    decoder: 1e-3
+                optim:
+                    lr: 3e-3   
+        """
+        if not hasattr(self, "parameters"):
+            self._optimizer_param_groups = None
+            return
+
+        known_groups = []
+        param_groups = []
+        if hasattr(self.cfg, "optim_param_groups"):
+            param_groups_cfg = self.cfg.optim_param_groups
+            for group, lr in param_groups_cfg.items():
+                module = getattr(self, group, None)
+                if module and hasattr(module, "parameters"):
+                    params = module.parameters()
+                    known_groups.append(group)
+                    param_groups.append({"params": params, "lr": lr})
+
+            other_params = []
+            for n, p in self.named_parameters():
+                is_unknown = True
+                for group in known_groups:
+                    if n.startswith(group):
```

Comment:
This makes an assumption that Nemo models are not nested which is incorrect. We can have Nemo model inside Nemo model and so it's param state dict will change (inner model . Group)

Rewrite using better logic to determine params. 
Also note, nested Nemo models can have more than one group - two encoders from two different models.

Reply:
If it's a nested model, we can use some config like
```
optim_param_groups:
    "module1.encoder": 1e-3
    "module2.encoder": 1e-4
```
Do you think this can handle the situation?